openmmtools/integrators.py

```diff
@@ -746,5 +771,19 @@ def __init__(self, temperature=298.0 * simtk.unit.kelvin, collision_rate=91.0 /
         #
         # Velocity randomization
         #
+        if monitor_heat:
+            self.addComputeSum("kinetic_energy_2", "0.5 * m * v * v")
+        
         self.addComputePerDof("v", "sqrt(b)*v + sqrt(1-b)*sigma*gaussian")
         self.addConstrainVelocities()
+        
+        if monitor_heat:
+            self.addComputeSum("kinetic_energy_3", "0.5 * m * v * v")
+        
+        if monitor_heat:
+            self.addComputeGlobal("heat", "(kinetic_energy_1 - kinetic_energy_0) + (kinetic_energy_3 - kinetic_energy_2)")
+        
+        if monitor_work:
+            # accumulate shadow_work, where W_step = ∆E_step - Q_step
+            self.addComputeGlobal("new_energy", "kinetic_energy_3 + energy")
+            self.addComputeGlobal("shadow_work", "shadow_work + (new_energy - old_energy) - heat")
```

Comment:
Ahh, good point! I had just implemented it that way following this comment in the To-Do's: https://github.com/choderalab/openmmtools/blob/0.8.0/openmmtools/integrators.py#L683 , but I think the alternative you suggest is simpler, and also lets me control `monitor_work` and `monitor_heat` separately.